File: backend/messenger/src/main/java/com/cloudapp/messenger/MessengerApplication.java
package com.cloudapp.messenger;

import javax.servlet.ServletContextEvent;
import javax.servlet.ServletContextListener;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

@SpringBootApplication
public class MessengerApplication {
	private static Log logger = LogFactory.getLog(MessengerApplication.class);

	@Bean
	protected ServletContextListener listener() {
		return new ServletContextListener() {

			@Override
			public void contextInitialized(ServletContextEvent sce) {
				logger.info("ServletContext initialized");
			}

			@Override
			public void contextDestroyed(ServletContextEvent sce) {
				logger.info("ServletContext destroyed");
			}

		};
	}
	public static void main(String[] args) {
		SpringApplication.run(MessengerApplication.class, args);
	}

}

File: backend/messenger/src/main/java/com/cloudapp/messenger/MessengerApplication.kt
package com.cloudapp.messenger

import org.apache.commons.logging.LogFactory
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import javax.servlet.ServletContextEvent
import javax.servlet.ServletContextListener

@SpringBootApplication
class MessengerApplication {
    @Bean
    protected fun listener(): ServletContextListener {
        return object : ServletContextListener {
            override fun contextInitialized(sce: ServletContextEvent) {
                logger.info("ServletContext initialized")
            }

            override fun contextDestroyed(sce: ServletContextEvent) {
                logger.info("ServletContext destroyed")
            }
        }
    }

    companion object {
        private val logger = LogFactory.getLog(
            MessengerApplication::class.java
        )

        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(MessengerApplication::class.java, *args)
        }
    }
}